### ISSUE TITLE
Arrows shouldn't bind to any shapes

### ISSUE THREAD

**Issue by smyrman** [excalidraw/excalidraw#4797]
![output](https://user-images.githubusercontent.com/75205/154266289-96bdd9ce-92c9-49c3-a3be-b25fe6ba850f.gif)

I love Excalidraw low fidelity approach, but the arrow bindings are driving me insane. Arrows locks to _everything_, and as you edit text, resize elements, move elements, the arrows _always_ end up wrong once the operation is complete.

Related to #4324, I want to suggest that arrows should not bind to any shapes _at all_, and let them behave like any other object, rather than _trying_ to be smart about it.

Count the "time saved" by arrows locking to objects v.s "time lost" by arrows locking to objects, at least for me that would be something like seconds saved and hours lost when aggregated.

**Comment by smyrman** [comment]
PS! I know it's _hard_ to remove something. According to recent studies, it's _against human nature_. But in this case, I think removing the feature of arrow binding _completely_ is the best solution for the product.

> Adding is favoured over subtracting in problem solving
https://www.nature.com/articles/d41586-021-00592-0

**Comment by dwelle** [comment]
> In this case, I really think removing the feature of arrow binding completely is the best solution for the product, and in-line with the overall focus on low fidelity.

I agree. There are definitely plans to scale it back, or at worst make it easy to disable it.

**Comment by DanTulovsky** [comment]
As a counterpoint, I would be very sad if arrows were just their own objects. e.g. If I move a box with an attached arrow, I want the arrow to move with the box.

**Comment by smyrman** [comment]
> As a counterpoint, I would be very sad if arrows were just their own objects. e.g. If I move a box with an attached arrow, I want the arrow to move with the box.

@DanTulovsky, how often do you experience moving a box, and _not_ ending up adjusting the arrow manually afterwards anyway? And if you do end up adjusting the arrow afterwards, the question is, how much time did the automatic moving of the arrow save you?

Some products that specialize in drawing boxes with arrows between them, e.g. draw.io or various Office variants/clonse, have used a lot if time in specifying different _snap points_ along the objects, and defining different types of curve adjustments, so that when you move the box, the arrow ends up "right" afterward. Excalidraw (rightfully IMO) haven't done this.

This is my experience roughly 90% of the time if I want to move a box when there is an arrow pointing at it:

![output](https://user-images.githubusercontent.com/75205/154434707-8c16a9ea-ca89-438d-bbe7-e3bc571be6e8.gif)

**Comment by smyrman** [comment]
For comparison without snapping:
![output](https://user-images.githubusercontent.com/75205/154437829-eb73d748-6c89-43cc-bab5-d290355f35d8.gif)

**Comment by DanTulovsky** [comment]
I think the answer to your question is "sometimes", but I still prefer the current behavior.  A toggle as proposed above seems like a reasonable solution.

**Comment by theol0403** [comment]
The snapping is driving me insane, and wastes a lot of time to work around. Whenever I have any text near an arrow, I cannot adjust the text without the arrow going insane.

I vote for opt-in for the feature, and also possibly to reduce the number of things arrows can snap to. Would it ever be useful to snap an arrow to an element of some text? Maybe removing snapping to text altogether would be good.

Another approach could be a feature where whenever a snapping possibility is detected, a little button shows up near the potential snap to allow the user to "lock" the elements together. This way it allows the feature to present itself whenever it could be useful, rather than having a global toggle that the user either leaves on and it causes problems, or leaves off and the user never remembers to use it.

**Comment by baldurh** [comment]
My two cents:

- Arrows should snap to shapes, not text
- Text should be able to snap with arrows (eg. for description text on arrows)
- Maybe introduce anchor points on objects for when you don’t want to change the position of the snap when you move stuff around

**Comment by vim-usds** [comment]
Love excalidraw! Thank you for an awesome product!

+100 points for everything awesome. -10 for auto-snapping. Looking forwarding to any solution that allows it to not happen.

**Comment by qodesmith** [comment]
Is there currently a way to "unbind" an arrow?

**Comment by theol0403** [comment]
> Is there currently a way to "unbind" an arrow?

The only way is to cut and paste. Sucks that this issue has not yet been resolved.

**Comment by dwelle** [comment]
I feel everyone in this thread. I wonder what to do short of 1) disabling arrow-text binding altogether 2) make it opt-in/opt-out. As suggested somewhere above, (2) will be likely the way to go, but it's not really a solution as the default behavior (which will likely remain to be bind-by-default) will still not work properly.

**Comment by baldurh** [comment]
@smyrman wrote:
> > As a counterpoint, I would be very sad if arrows were just their own objects. e.g. If I move a box with an attached arrow, I want the arrow to move with the box.
> 
> @DanTulovsky, how often do you experience moving a box, and _not_ ending up adjusting the arrow manually afterwards anyway? And if you do end up adjusting the arrow afterwards, the question is, how much time did the automatic moving of the arrow save you?
> 
> Some products that specialize in drawing boxes with arrows between them, e.g. draw.io or various Office variants/clonse, have used a lot if time in specifying different _snap points_ along the objects' edge, and defining different types of curve adjustments, so that when you move the box, the arrow ends up "right" afterward. Excalidraw (rightfully IMO) haven't spent _as much_ effort on this.
> 
> This is my experience roughly 90% of the time if I want to move a box, and have the arrow follow:
> 
> ![output](https://user-images.githubusercontent.com/75205/154434707-8c16a9ea-ca89-438d-bbe7-e3bc571be6e8.gif) 

I know your comment is pretty old and it seems the behaviour has changed a bit. But I don’t remember ever having this particular problem. What you’re basically doing is unbinding and rebinding the arrow when you attempt to move the right endpoint and then rebinding the left endpoint. This way you’re doing an unnecessary extra step. Here’s how it looks for me right now.

![excalidraw](https://user-images.githubusercontent.com/1823617/188412639-ec1ba9ee-5885-4509-aa3c-d104c24c742c.gif)

But I just want to add that I regularly work with big and complex diagrams with lots of boxes and multiple arrows connecting to them. I’d be devastated if arrow binding would not work anymore. It would just make everything so much worse. Even if the arrows don’t end up in the perfect place 100% of the time it’s still better to let the arrows follow the bound object.

Example with binding:
![excalidraw2](https://user-images.githubusercontent.com/1823617/188415078-e1cc8c5e-f6c1-4644-b0bd-e1481adefe70.gif)

Example without binding (forgot to move one of the boxes but you get the idea):
![excalidraw3](https://user-images.githubusercontent.com/1823617/188415122-bf177f64-ea96-49c3-a848-1e7165f33b47.gif)

**Comment by baldurh** [comment]
Could the solution be one of the following?
A) Add a configuration. Enable/disable binding? Any arrow added/moved with this disabled would not bind to anything.
B) Add a configuration. Something like ‘Break binding when moving objects’ so that if you want to move something and ignore all bindings, this is something you could turn on.
C) Add a hotkey. Holding down some key while moving an object would ignore all bound arrows and just move the object. Or a hotkey for toggling option B

**Comment by ad1992** [comment]
Closing this as I have created a new issue where we will be discussing this and improve the overall UX for arrow binding https://github.com/excalidraw/excalidraw/issues/6685
